// This is the public API offered by your entity.
syntax = "proto3";

import "google/protobuf/empty.proto";
import "akkaserverless/annotations.proto";
import "google/api/annotations.proto";

import "thing_domain.proto";

package com.example.eventsourced;


message Thing {
  string thing_id = 1 [(akkaserverless.field).entity_key = true];
  string name = 2;
}

message GetThingCommand {
  string thing_id = 1 [(akkaserverless.field).entity_key = true];
}

service ThingEventSourcedService {

    rpc AddThing(Thing) returns (google.protobuf.Empty) {
        option (google.api.http) = {
        post: "/event_sourced/things"
            body: "*"
        };
    }

    rpc GetThing(GetThingCommand) returns (ThingState) {
        option (google.api.http) = {
        get: "/event_sourced/things/{thing_id}"
        };
    }

    
}

